java/source/dataset.rst

```diff
@@ -0,0 +1,232 @@
+.. _arrow-dataset:
+
+=======
+Dataset
+=======
+
+Current java project that use JNI are:
+
+* `Arrow Java Dataset <https://arrow.apache.org/docs/dev/java/dataset.html>`_: Java implementation of Arrow Dataset API/Framework. JniLoader [arrow_cdata_jni]
+* `Arrow Java C Data Interface <https://arrow.apache.org/docs/format/CDataInterface.html>`_: Java implementation of C Data Interface. JniLoader [arrow_dataset_jni]
+
+.. contents::
+
+Schema
```

Comment:
I think we should reorganize this a bit. This first part is about constructing a dataset, not about schemas per se. 

When constructing a dataset, we can pass in one or more files and a file format. A dataset needs a schema. We can explicitly ask the dataset factory to inspect the files and return a schema. We should still construct the dataset with `DatasetFactory.finish(schema).` That's the first recipe.

We can add a recipe about letting the factory infer the schema. We don't call `inspect` and just call `finish`.

Then the second recipe can become "Getting the schema of a dataset". The JNI bindings don't expose this except by starting a scan, so we demonstrate that.

Reply:
We should also add a recipe where we construct a factory from more than one file.

Reply:
Added a recipe for dataset
Added a recipe for construct a factory for a directory reading